website/server/controllers/api-v4/auth.js

```diff
@@ -83,4 +86,37 @@ api.registerLocal = {
   },
 };
 
+/**
+ * @api {put} /api/v3/user/auth/check-email Check if email is used
+ * @apiDescription Check if the email is already used by another user
+ * @apiName CheckEmail
+ * @apiGroup User
+ *
+ * @apiParam (Body) {String} email The checked email address.
+ *
+ * @apiSuccess {String} data.email The checked email address
+ */
+api.checkEmail = {
+  method: 'POST',
+  url: '/user/auth/check-email',
+  async handler (req, res) {
+    req.checkBody({
+      email: {
+        notEmpty: { errorMessage: res.t('missingEmail') },
+      },
+    });
+
+    const validationErrors = req.validationErrors();
+    if (validationErrors) throw validationErrors;
+
+    const emailAlreadyInUse = await User.findOne({
+      'auth.local.email': req.body.email.toLowerCase(),
```

Comment:
This catches the scenario where the email exists in the database as lowercase, and the request is mixed case. What about the other way around? We might need to do a case insensitive _search_ rather than simply lowercase the input.

Reply:
Both during registration as well as when updating the an email address, the email is already set to lowercase. A case insensitive search should be significantly slower and since the database should already contain only lowercase emails, it should not be an issue I think?